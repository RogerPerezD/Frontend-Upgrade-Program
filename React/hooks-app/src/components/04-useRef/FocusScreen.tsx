import React, { useRef } from 'react';

export const FocusScreen = () => {
    const inputRef = useRef<HTMLInputElement>(null);

    const handleClick  = () =>{
        inputRef.current?.select();
    }

  return (<div>
      <h1>Focus Screen</h1>

      <hr />

      <input 
      type="text"
      className='form-control'
      placeholder='Su nombre' 
      ref={ inputRef }
      />

    <button 
    className='btn btn-primary mt-3'
    onClick={ handleClick }>Click</button>

  </div>);
};
