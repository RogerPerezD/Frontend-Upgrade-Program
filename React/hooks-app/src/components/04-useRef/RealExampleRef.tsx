import React, { useState } from 'react';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';


export const RealExampleRef = () => {

    const [show, setShow] = useState<boolean>(false);

    const handleClick = () =>{
        setShow( !show );
    }

    return (
        <>
        <h1>Real Example</h1>
        <hr />
        { show && <MultipleCustomHooks/>}
        <button className='btn btn-primary mt-5' 
        onClick={ handleClick }>Hide/Show</button>
        </>
    );
};
