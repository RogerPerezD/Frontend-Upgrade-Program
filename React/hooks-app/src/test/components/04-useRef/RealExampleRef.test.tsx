import { shallow } from "enzyme";
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";

describe('Test un <RealExampleRef/>', () => {
    
    test('should show the component succesfully', () => {
        
        const wrapper = shallow(<RealExampleRef/>);

        expect( wrapper ).toMatchSnapshot();
    });

    test('should return the component <MultipleCustomHooks/> inside our component', () => {
        const wrapper = shallow(<RealExampleRef/>);

        const button = wrapper.find('button');
        button.simulate('click');

        const component = wrapper.find('MultipleCustomHooks');

        console.log(component);
        expect( component.exists() ).toBe( true );
    });
});