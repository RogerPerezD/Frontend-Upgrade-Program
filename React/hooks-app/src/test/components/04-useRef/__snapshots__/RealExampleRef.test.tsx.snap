// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test un <RealExampleRef/> should show the component succesfully 1`] = `
<Fragment>
  <h1>
    Real Example
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Hide/Show
  </button>
</Fragment>
`;
